pkg/providers/common/instancetype/instancetype.go

```
@@ -741,24 +744,64 @@ func (p *IBMInstanceTypeProvider) convertVPCProfileToInstanceType(ctx context.Co
 			corev1.ResourcePods:   *podResource,
 			"nvidia.com/gpu":      *gpuResource, // Standard Kubernetes GPU resource name
 		},
-		Overhead: &cloudprovider.InstanceTypeOverhead{
-			KubeReserved: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("100m"),
-				corev1.ResourceMemory: resource.MustParse("1Gi"),
-			},
-			SystemReserved: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("100m"),
-				corev1.ResourceMemory: resource.MustParse("1Gi"),
-			},
-			EvictionThreshold: corev1.ResourceList{
-				corev1.ResourceMemory: resource.MustParse("500Mi"),
-			},
-		},
+		Overhead:     overhead,
 		Requirements: requirements,
 		Offerings:    offerings,
 	}, nil
 }
 
+func (p *IBMInstanceTypeProvider) calculateOverhead(nodeClass *v1alpha1.IBMNodeClass) *cloudprovider.InstanceTypeOverhead {
+	// Default values if kubelet config not specified
+	kubeReservedCPU := resource.MustParse("100m")
+	kubeReservedMemory := resource.MustParse("1Gi")
+	systemReservedCPU := resource.MustParse("100m")
+	systemReservedMemory := resource.MustParse("1Gi")
+	evictionThreshold := resource.MustParse("500Mi")
+
+	// Override with user-specified values
+	if nodeClass != nil && nodeClass.Spec.Kubelet != nil {
+		if cpu, ok := nodeClass.Spec.Kubelet.KubeReserved["cpu"]; ok {
+			if parsed, err := resource.ParseQuantity(cpu); err == nil {
+				kubeReservedCPU = parsed
+			}
+		}
+		if mem, ok := nodeClass.Spec.Kubelet.KubeReserved["memory"]; ok {
+			if parsed, err := resource.ParseQuantity(mem); err == nil {
+				kubeReservedMemory = parsed
+			}
+		}
+		if cpu, ok := nodeClass.Spec.Kubelet.SystemReserved["cpu"]; ok {
+			if parsed, err := resource.ParseQuantity(cpu); err == nil {
+				systemReservedCPU = parsed
+			}
+		}
+		if mem, ok := nodeClass.Spec.Kubelet.SystemReserved["memory"]; ok {
+			if parsed, err := resource.ParseQuantity(mem); err == nil {
+				systemReservedMemory = parsed
+			}
+		}
```

this looks to me like invalid user input (e.g., `"not-a-quantity"`) silently falls back to defaults and thus users get no feedback their config is wrong